error_check_logic.py

```diff
@@ -56,6 +55,21 @@ def Run(self, institution, session):
     self._Validate(class_groups, isValid_class_groups, 'Class Groups')
     self._Validate(student_groups, isValid_student_groups, 'Student Groups')
 
+    # Only call yaml.load if data is valid. And store the result
```

Comment:
Actually the schema could be changed to remember the loaded yaml so we don't have to load it again below.